docs/tutorial/index.org

```diff
@@ -0,0 +1,1112 @@
+# (venv) enos:~/enos-myxp$
+openstack server show cli-vm
+#+END_SRC
+
+Note in particular the status of your VM. This status will go from
+~BUILD~: OpenStack is looking for the best place to boot the VM, to
+~ACTIVE~: your VM is running. The status could also be ~ERROR~ if you
+are experiencing hard times with your infrastructure.
+
+With the previous ~openstack server create~ command, the VM boots with
+a private IP. Private IPs are used for communication between VMs,
+meaning you cannot ping your VM from the lab machine. Network lovers
+will find a challenge here: try to ping the VM from the lab machine.
+For the others, you have to manually affect a floating IP to your
+machine if you want it pingable from the enos node.
+
+#+BEGIN_SRC sh :tangle ../../tests/functionnal/tests/tutorial/enos-node.sh
+# (venv) enos:~/enos-myxp$
+openstack server add floating ip\
+  cli-vm\
+  $(openstack floating ip create public -c floating_ip_address -f value)
+#+END_SRC
+
+You can ask for the status of your VM and its IPs with:
+#+BEGIN_SRC sh :tangle ../../tests/functionnal/tests/tutorial/enos-node.sh
+# (venv) enos:~/enos-myxp$
+openstack server show cli-vm -c status -c addresses # (venv) enos-node$
+#+END_SRC
+
+Wait one minute or two the time for the VM to boot, and when the state
+is ~ACTIVE~, you can ping it on its floating IP and SSH on it:
+
+#+BEGIN_SRC sh
+# (venv) enos:~/enos-myxp$
+ping <floating-ip>
+# (venv) enos:~/enos-myxp$
+ssh -l cirros <floating-ip>
+#+END_SRC
+
+#+BEGIN_note
+Waiting for the IP to appear and then ping it could be done with a
+bunch of bash commands, such as in listing [[lst:query-ip]].
+
+#+CAPTION: Query the ~cli-vm~, wait until the floating IP is affected,
+#+CAPTION: and finally ping it.
+#+NAME: lst:query-ip
+#+BEGIN_SRC sh :tangle ../../tests/functionnal/tests/tutorial/enos-node.sh
+FLOATING_IP="openstack server show cli-vm -c addresses -f value | sed  -Er 's/private=(10\.0\.0\.[0-9]+).*/\1/g'"
+while [ -z "$(eval $FLOATING_IP)" ]; do
+  sleep 1
+done
+ping -c 3 "$(eval $FLOATING_IP)"
+#+END_SRC
+
+You can also check that the VM finished to boot by looking at its logs
+with ~openstack console log show cli-vm~. The cirros VM finished to
+boot when last lines are:
+#+BEGIN_EXAMPLE
+=== cirros: current=0.3.4 uptime=16.56 ===
+  ____               ____  ____
+ / __/ __ ____ ____ / __ \/ __/
+/ /__ / // __// __// /_/ /\ \
+\___//_//_/  /_/   \____/___/
+   http://cirros-cloud.net
+
+
+login as 'cirros' user. default password: 'cubswin:)'. use 'sudo' for root.
+cli-vm login:
+#+END_EXAMPLE
+#+END_note
+
+Before going to the next section, play around with the ~openstack~ CLI
+and Horizon. For instance, list all the features offered by Nova with
+~openstack server --help~. Here are some commands:
+1. SSH on ~cli-vm~ using its name rather than its private IP.
+   #+BEGIN_SRC sh :tangle ../../tests/functionnal/tests/tutorial/enos-node.sh
+   # (venv) enos:~/enos-myxp$
+   openstack server ssh cli-vm\
```

Comment:
The correct syntax seems to be: 
`openstack server ssh cli-vm --public --login cirros --option BatchMode=yes --option ConnectionAttempts=1`

Reply:
We shouldn't not tangled the ssh command as it will block.

Another solution would be to use a pubkey authentification and test the ssh connection.

Reply:
I use the `noweb` syntax of org (with `:noweb strip-export`) to include the pubkey authentication in the functional test while not including it in the tp.